src/kernel-install/90-loaderentry.install

```diff
@@ -21,45 +21,45 @@ MACHINE_ID=$KERNEL_INSTALL_MACHINE_ID
 ENTRY_DIR="/$MACHINE_ID/$KERNEL_VERSION"
 BOOT_ROOT=${ENTRY_DIR_ABS%$ENTRY_DIR}
 
-if [[ $COMMAND == remove ]]; then
+if [ "$COMMAND" = remove ]; then
     rm -f "$BOOT_ROOT/loader/entries/$MACHINE_ID-$KERNEL_VERSION.conf"
     rm -f "$BOOT_ROOT/loader/entries/$MACHINE_ID-$KERNEL_VERSION+"*".conf"
     exit 0
 fi
 
-if ! [[ $COMMAND == add ]]; then
+if ! [ "$COMMAND" = add ]; then
     exit 1
 fi
 
-if ! [[ $KERNEL_IMAGE ]]; then
+if [ -z "$KERNEL_IMAGE" ]; then
     exit 1
 fi
 
-if [[ -f /etc/os-release ]]; then
+if [ -f /etc/os-release ]; then
     . /etc/os-release
-elif [[ -f /usr/lib/os-release ]]; then
+elif [ -f /usr/lib/os-release ]; then
     . /usr/lib/os-release
 fi
 
-if ! [[ $PRETTY_NAME ]]; then
+if [ -z "$PRETTY_NAME" ]; then
     PRETTY_NAME="Linux $KERNEL_VERSION"
 fi
 
-if [[ -f /etc/kernel/cmdline ]]; then
-    read -r -d '' -a BOOT_OPTIONS < /etc/kernel/cmdline
-elif [[ -f /usr/lib/kernel/cmdline ]]; then
-    read -r -d '' -a BOOT_OPTIONS < /usr/lib/kernel/cmdline
-else
-    declare -a BOOT_OPTIONS
+if [ -f /etc/kernel/cmdline ]; then
+    read -r BOOT_OPTIONS < /etc/kernel/cmdline
+elif [ -f /usr/lib/kernel/cmdline ]; then
+    read -r BOOT_OPTIONS < /usr/lib/kernel/cmdline
+fi
 
-    read -r -d '' -a line < /proc/cmdline
-    for i in "${line[@]}"; do
-        [[ "${i#initrd=*}" != "$i" ]] && continue
-        BOOT_OPTIONS+=("$i")
-    done
+if [ -z "${BOOT_OPTIONS}" -a -f /proc/cmdline ]; then
+    BOOT_OPTIONS="$(for i in $(cat /proc/cmdline)
+        do
+            [ "${i#initrd=*}" != "$i" ] || echo $i
```

Comment:
This creates a newline-separated list. The line with "options" below becomes multi-line, syntactically invalid. `echo -n` should work, but please make check that the output looks OK.

Reply:
Why do you think it’s syntactically invalid? Multi-line double-quoted strings are perfectly fine in shell.

Reply:
Oh, I see what you mean.